com.unity.render-pipelines.high-definition/Runtime/Material/Decal/DecalProjector.cs

```diff
@@ -224,6 +262,45 @@ internal DecalSystem.DecalHandle Handle
             }
         }
 
+        // Struct used to gather all decal property required to be cached to be sent to shader code
+        internal struct CachedDecalData
+        {
+            public Matrix4x4 localToWorld;
+            public Quaternion rotation;
+            public Matrix4x4 sizeOffset;
+            public float drawDistance;
+            public float fadeScale;
+            public float startAngleFade;
+            public float endAngleFade;
+            public Vector4 uvScaleBias;
+            public bool affectsTransparency;
+            public int layerMask;
+            public ulong sceneLayerMask;
+            public float fadeFactor;
+            public DecalLayerEnum decalLayerMask;
+        }
+
+        internal CachedDecalData GetCachedDecalData()
```

Comment:
Is there any way we can actually cache that and not reconstruct every time we need it?  (might not be, not super familiar with the rest of the code)

Reply:
I am not sure, guess when we call this function it usually mean that there is a change (in particular the rotation matrix), but I can investigate)